cleanlab_studio/cli/api_service.py

```diff
@@ -187,6 +143,117 @@ async def upload_files_for_filepath_column(filepath_column: str) -> None:
         handle_api_error_from_json(json.loads(res_text))
 
 
+async def upload_files_for_filepath_column(
+    session: aiohttp.ClientSession,
+    api_key: str,
+    dataset_id: str,
+    dataset_filepath: str,
+    rows: List[List[Any]],
+    row_ids: List[Any],
+    columns: List[str],
+    filepath_column: str,
+    media_type: str,
+) -> None:
+    """
+    Uploads all files in a filepath column to S3
+
+    :param session: client session for making http requests
+    :param api_key: api key for getting presigned posts
+    :param dataset_id: id of dataset to upload files for
+    :param dataset_filepath: filepath of dataset to upload files for (can be empty string for simple image upload)
+    :param rows: dataset rows
+    :param row_ids: ids of dataset rows
+    :param columns: dataset column names
+    :param filepath_column: name of the column to upload files for
+    :param media_type: type of media to upload
+    """
+    sem = asyncio.Semaphore(MAX_PARALLEL_UPLOADS)
+    cancelled = asyncio.Event()
+    filepath_column_idx = columns.index(filepath_column)
+    filepaths = [row[filepath_column_idx] for row in rows]
+    dataset_dir: pathlib.Path = pathlib.Path(dataset_filepath).parent
+    absolute_filepaths = [
+        str(get_image_filepath(dataset_dir, row[filepath_column_idx])) for row in rows
+    ]
+    filepath_to_posts = get_presigned_posts(
+        api_key=api_key,
+        dataset_id=dataset_id,
+        filepaths=filepaths,
+        row_ids=row_ids,
+        media_type=media_type,
+    )
+    for coro in asyncio.as_completed(
+        [
+            post_file(
+                session,
+                original_filepath,
+                absolute_filepath,
+                filepath_to_posts.get(original_filepath),
```

Comment:
we want to fail hard here if this key doesn't exist, instead of continuing with a `None` post_data

Reply:
Do we want all the file uploads to fail if the key doesn't exist? Or just the upload for the file whose key is missing

Reply:
we currently fail hard if any file fails to upload, so I think it should probably be the same here.

we don't have any resiliency to missing media in training afaik, so if we allowed the upload to continue with some missing data here, we'd run into issues at train time